### DISCUSSION TITLE
Roadmap for MA identifiers

### DISCUSSION THREAD

**Discussion by mihai-sysbio** [MetabolicAtlas/MetabolicAtlas#1388] [Ideas]
While the current process of keeping track of identifiers has worked until now, a better approach should be investigated. Please contribute!

**Facts**
Currently, MA identifiers are coined and kept track of in the [Human-GEM repository](https://github.com/SysBioChalmers/Human-GEM). The identifiers in use are listed in the model files, in particular [reactions.tsv](https://github.com/SysBioChalmers/Human-GEM/blob/main/model/reactions.tsv) and [metabolites.tsv](https://github.com/SysBioChalmers/Human-GEM/blob/main/model/metabolites.tsv), while the deprecated identifiers are moved over in the [corresponding files](https://github.com/SysBioChalmers/Human-GEM/tree/main/data/deprecatedIdentifiers). There are also GH Actions in this repository that check the consistency of the identifier use/reuse.

GEMs templated from Human-GEM also [use MA identifiers, which can change](https://github.com/SysBioChalmers/Mouse-GEM/commit/6dc2a71237a01cebef03f957ca884d34436be05f).

**Concerns**
As a user of Metabolic Atlas, I want to easily find deprecated identifiers for my version of my model, so that I can understand what changed across versions.
As a model curator, I want to coin new identifiers easily, so that I can introduce new reactions/metabolites.
As a model curator, I want to deprecate identifiers easily, so that I can implement the curation work.

**Opportunities**
There have been discussions about how to keep track of merged entities here https://github.com/SysBioChalmers/Human-GEM/discussions/615

**Vision**
It's likely that the optimal solution includes setting up a centralised repository for the identifiers.

**Comment by haowang-bioinfo** [comment]
now it does appear to a good time point to revisit this MA id issue.

> GEMs templated from Human-GEM also [use MA identifiers, which can change](https://github.com/SysBioChalmers/Mouse-GEM/commit/6dc2a71237a01cebef03f957ca884d34436be05f).

so far Human-GEM and five model animal GEMs have been using the MA ids, whose wider application is expected

> It's likely that the optimal solution includes setting up a centralised repository for the identifiers.

agree with the idea of setting up a centralised repository, to which the migration plan for MA ids from Human-GEM may be discussed

**Reply by mihai-sysbio** [reply]
@haowang-bioinfo  do you have any ideas for how to address the concerns put forth? or perhaps other concerns which are not mentioned?

> As a user of Metabolic Atlas, I want to easily find deprecated identifiers for my version of my model, so that I can understand what changed across versions.
> As a model curator, I want to coin new identifiers easily, so that I can introduce new reactions/metabolites.
> As a model curator, I want to deprecate identifiers easily, so that I can implement the curation work.

**Reply by haowang-bioinfo** [reply]
to move forward, a follow-up plan may begin with considering the migration of existing GH actions surrounding MA ids from HumanGEM to a dedicated and centralised repo

**Reply by mihai-sysbio** [reply]
> to move forward, a follow-up plan may begin with considering the migration of existing GH actions surrounding MA ids from HumanGEM to a dedicated and centralised repo

Indeed, that's one of the things to do. However, it does not solve any of the concerns above. And without a plan to tackle these, I'm not keen on introducing changes in a (somewhat) functional system (that we have now for the identifiers).

**Comment by mihai-sysbio** [comment]
@smoretti and @mpagni12 you probably have some experience with this, any advice?

**Reply by smoretti** [reply]
This is not an easy task. In MetaNetX we have a specific file to trace deprecated entries:
- their id
- what they became (merge, split, ...)
- when they changed (at which release)
A graph view or the SPARQL endpoint help to follow their history.

I think the main point is that deprecated ids must not disappear, to always find what they became, and to avoid to reuse them.

Unfortunately I think there is no specific qualifier in SBML to tag them.
https://github.com/combine-org/combine-specifications/blob/main/specifications/qualifiers.md

**Comment by mihai-sysbio** [comment]
@edkerk any potential concerns coming from the perspective of `yeast-GEM`, which is not using MA identifiers?

**Reply by edkerk** [reply]
No concerns, but I like recognize your concerns and share your vision. It would be preferred to also apply MA identifiers in `yeast-GEM`, but I have no suggestions on how to exactly setup such a centralised repository.

**Comment by mihai-sysbio** [comment]
On the same topic, @smoretti has also asked if MetAtlas has a centralized listing of all the identifiers. I believe this would be useful, and doable. @smoretti it would be great if you could expand on what would need to be included, or if the raw list of active (non-deprecated) identifiers is sufficient.

**Reply by mihai-sysbio** [reply]
There was also a question about compartments. For those, there are no identifiers, but we are using the naming convention agreed upon here https://github.com/SysBioChalmers/Human-GEM/pull/341 . Perhaps that's a new feature request - compartment identifiers?

**Comment by smoretti** [comment]
For compounds, ideally, we would like to have ids, names, cross-references and structures
     + synonyms, formula, mass and charge if possible.

For reactions, ids, equations and cross-references
     + names, synonyms, EC, reversibility and pathways if possible.
